src/fknni/faiss/faiss.py

```diff
@@ -32,87 +31,172 @@ def __init__(
                       where closer neighbors have a higher influence on the imputation.
             index_factory: Description of the Faiss index type to build. Defaults to 'Flat'.
         """
-        super().__init__()
+        if n_neighbors < 1:
+            raise ValueError("n_neighbors must be at least 1.")
+        if strategy not in {"mean", "median", "weighted"}:
+            raise ValueError("Unknown strategy. Choose one of 'mean', 'median', 'weighted'")
+
         self.missing_values = missing_values
         self.n_neighbors = n_neighbors
         self.metric = metric
         self.strategy = strategy
         self.index_factory = index_factory
+        self.min_data_ratio = 0
```

Comment:
We're adding a bunch of additional attributes here but I don't see all of them being set later or passable as parameters here. Can we make all of them configurable and ensure that they have a docstring somewhere, please?